src/react/backend/location/LocationDetails/LocationDetailsTapeDMF.tsx

```diff
@@ -0,0 +1,186 @@
+  password: '',
+};
+
+const RepoIdInputContainer = styled.div`
+  display: flex;
+  gap: 1rem;
+  align-items: center;
+`;
+
+function RepoIdsInput({
+  repoIds,
+  onChange,
+}: {
+  repoIds: string[];
+  onChange: (repoIds: string[]) => void;
+}) {
+  const insertEntry = () => {
+    onChange([...repoIds, '']);
+  };
+
+  const deleteEntry = (entryIndex: number) => {
+    let tempRepoIds = [...repoIds];
+    tempRepoIds.splice(entryIndex, 1);
+    if (tempRepoIds.length === 0) {
+      tempRepoIds = [''];
+    }
+    onChange([...tempRepoIds]);
+  };
+
+  return (
+    <>
+      {
+        <Fieldset>
+          <Label required htmlFor={`repoIds[${repoIds.length - 1}]`}>
+            RepoId(s)
+          </Label>
+          {repoIds.map((repoId, index) => (
+            <RepoIdInputContainer key={index}>
+              <Input
+                name={`repoIds[${index}]`}
+                id={`repoIds[${index}]`}
+                type="text"
+                placeholder=""
+                value={repoId}
+                onChange={(evt) => {
+                  const tempRepoIds = [...repoIds];
+                  tempRepoIds[index] = evt.target.value;
+                  onChange(tempRepoIds);
+                }}
+                autoComplete="off"
+              />
+              <SubButton
+                index={index}
+                key={`delete-${repoId}`}
+                deleteEntry={deleteEntry}
+                items={repoIds}
+                disabled={repoId === ''}
+              />
+              <AddButton
+                index={index}
+                key={`add-${repoId}`}
+                insertEntry={insertEntry}
+                items={repoIds}
+                disabled={repoId === ''}
+              />
+            </RepoIdInputContainer>
+          ))}
+        </Fieldset>
+      }
+    </>
+  );
+}
+
+export default function LocationDetailsTapeDMF({
+  details,
+  onChange,
+}: LocationDetailsFormProps) {
+  const [formState, setFormState] = useState<State>(() => ({
+    ...Object.assign({}, INIT_STATE, details),
+  }));
+
+  const onInternalStateChange = (key: string, value: any) => {
+    setFormState({ ...formState, [key]: value });
+
+    if (onChange) {
+      onChange({ ...formState, [key]: value });
+    }
+  };
+
+  const onFormItemChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const target = e.target;
+    const value = target.type === 'checkbox' ? target.checked : target.value;
+    onInternalStateChange(target.name, value);
+  };
+
+  return (
+    <div>
+      <Fieldset>
+        <Label>Temperature</Label>
+        <SpacedBox mt={2}>
+        <span>
+          <i className="fas fa-snowflake" /> Cold
+        </span>
+        </SpacedBox>
+      </Fieldset>
+      <Fieldset>
+        <Label required htmlFor="endpoint">
+          Endpoint
+        </Label>
+        <Input
+          name="endpoint"
+          id="endpoint"
+          type="text"
+          placeholder="example: ws://path.to.my.dmf"
```

Comment:
I guess we could drop `example:` since it's already a placeholder?

Reply:
OK. As discussed with @Cuervino , it's suggested to have the example: presented.